src/com/owncloud/android/lib/common/accounts/AccountUtils.java

```diff
@@ -107,6 +109,29 @@ public static String getBaseUrlForAccount(Context context, Account account)
 
 
     /**
+     * Extracts server url for a file
+     * @param context
+     * @param account
+     * @param fileId remote file id
+     * @return url server for file or null on failure
+     */
+    public static String getUrlForFile (Context context, Account account, String fileId)
```

Comment:
Sorry, not seen this requested change, I've noticed that this class contains many methods for building urls but all of them are more related with accounts instead of with files themselves.  So as you said, it's better to keep this part in the OC-app, not in the library.